### ISSUE TITLE
-f silently fails if base/frameworks/packet-filter isn't loaded

### ISSUE THREAD

**Issue by jsiwek** [zeek/zeek#158]
Moved from https://bro-tracker.atlassian.net/browse/BIT-1407

Created by Vern Paxson at 2015-05-30T15:30:45.970-0500:
> I know we've been through this before (though searching the tickets in Jira, I couldn't find the thread).  But to revisit: the "-f filter" option silently does nothing if base/frameworks/packet-filter isn't loaded (so the scenario here is using -b to suppress its automatic loading).  This can lead to seriously confusing behavior.  It would be preferable if there's either an error message indicating that the option won't be supported, or if it forced loading of packet-filter.

Comment by Robin Sommer at 2015-06-01T10:11:09.695-0500:
> 
> 
> 
> 
> Yeah, I can see that. I think the main problem is the interaction
> between the command-line option and script, something that's rare
> (i.e., that the command-line option is tight that closely to a script
> being loaded). I would actually suggest we remove the command-line
> option altogether and instead work with a global: "bro -i eth0
> PacketFilter::filter=XXXX" (I believe we have a global with that
> effect already, otherwise we could add it).
> 
> Robin
> 
> 

Comment by Seth Hall at 2015-06-01T10:19:09.748-0500:
> 
> 
> We could alternately take the additional step of moving command line argument processing into Bro scripts.  I don’t know how exactly we’d solve this particular problem, but it should be reasonably solvable that way.
> 
>   .Seth
> 
> --
> Seth Hall
> International Computer Science Institute
> (Bro) because everyone has a network
> http://www.bro.org/
> 
> 

Comment by Robin Sommer at 2015-06-01T10:35:09.583-0500:
> 
> 
> 
> 
> If we had that, the '-f' switch would probably be added by the packet
> filter script, so it'd be available only when loaded.
> 
> I'd like to move argument processing into script-land as well.
> Generally, however, I think it would still be good to avoid arguments
> controlling script behaviour as much as possible. Main reason is
> argument inflation: -f has been historically around but there are
> plenty other scripts that in principle could take command-line
> arguments as well, which would get messy. And we have the X=Y syntax
> already to take care of that.
> 
> 

Comment by Vern Paxson at 2015-06-01T13:33:29.022-0500:
> While there's an appeal to processing arguments in script-land, because some arguments control basic script processing (e.g., -p), I'm not sure this can be done in a coherent fashion without some significant under-the-hood kludges.
> 
> Regarding replacing -f with PacketFilter::filter=XXXX, yuck - I wouldn't want to have to remember that, and there's no ready way for a user to discover this.
> 
> I'd be happy settling for -f warning (or exiting) with a statement that without the packet filtering framework loaded, it's a no-op.  Though I have a forboding that you're going to tell me that that's actually hard to implement :-P.
> 

Comment by Vern Paxson at 2015-06-01T13:35:33.426-0500:
> Also, regarding the policy script adding the -f flag: how would the user discover that that's what they need to do (have the policy script loaded) to get the flag?  I can see quite a bit of perplexment if Bro just tells the user that the flag doesn't exist without explaining why, if they don't have the script loaded.

**Comment by rsmmr** [comment]
@bmiki I see there's an older patch you did, is/was that working for you?

**Comment by bmiki** [comment]
Hi Robin,

Honestly speaking, I do not remember if I could test it or not.
I was interested in the zeek source code because I found a job offer by
Corelight. That's why I decided to dive into the source a bit, but I'm
totally newby.
As far as I remember I was able to reproduce the original problem and it
disappeared after applying the patch but I'm not completely sure if
I tested it the right way.

Br,
Miklos

Robin Sommer <notifications@github.com> ezt írta (időpont: 2020. aug. 12.,
Sze, 10:42):
